waffle_utils/video/tools.py

```diff
@@ -0,0 +1,163 @@
+        output_dir (Union[str, Path]): Path to the output directory where the frame images will be saved.
+        frame_rate (int, optional): Frame rate of the output images. Defaults to {DEFAULT_FRAME_RATE}.
+        output_image_extension (str, optional): Extension of the output frame images. Defaults to {DEFAULT_IMAGE_EXTENSION}.
+        verbose (bool, optional): Whether to print verbose output. Defaults to False.
+    """
+    input_path = Path(input_path)
+    output_dir = Path(output_dir)
+
+    if output_image_extension not in SUPPORTED_IMAGE_EXTENSION:
+        raise ValueError(
+            f"Invalid output_image_extension: {output_image_extension}.\n"
+            "Must be one of {SUPPORTED_IMAGE_EXTENSION}."
+        )
+
+    # Create output directory if it doesn't exist
+    if not output_dir.exists():
+        output_dir.mkdir(parents=True, exist_ok=True)
+
+    # Extract frames from the video file
+    video_capture = create_video_capture(input_path)
+    fps = video_capture.get(cv2.CAP_PROP_FPS)
+    success, image = video_capture.read()
+    count = 0
+    frame_interval = int(round(fps / frame_rate))
+    while success:
+        count += 1
+        if count % frame_interval == 0:
+            output_path = output_dir / f"{count}.{output_image_extension}"
+            save_image(output_path, image)
+
+            if verbose:
+                print(f"{input_path} ({count}) -> {output_path}.")
+
+        success, image = video_capture.read()
+
+    video_capture.release()
+    print(f"Output: {output_dir}/")
+
+
+def create_video(
+    input_dir: Union[str, Path],
+    output_path: Union[str, Path],
+    frame_rate: int = DEFAULT_FRAME_RATE,
+    verbose: bool = False,
+) -> None:
+    f"""Create a Video File from a Directory of Frame Images.
+
+    Args:
+        input_dir (Union[str, Path]): Path to the input directory containing the frame images.
+        output_path (Union[str, Path]): Path to the output video file.
+        frame_rate (int, optional): Frame rate of the output video. Defaults to {DEFAULT_FRAME_RATE}.
+        verbose (bool, optional): Whether to print verbose output. Defaults to False.
+    """
+    input_dir = Path(input_dir)
+    output_path = Path(output_path)
+
+    # Check if all files have the same extension
+    files = list(input_dir.glob("*"))
+    file_extensions = set()
+
+    for file in files:
+        file_extensions.add(file.suffix)
+
+    if len(file_extensions) != 1:
+        raise ValueError(
+            f"The files in {input_dir} do not have a consistent extension."
+        )
+
+    # Check if the extension of the files is supported
+    unique_extension = file_extensions.pop()[1:]  # Get the unique extension existing.
+    if unique_extension not in SUPPORTED_IMAGE_EXTENSION:
+        raise ValueError(
+            f"File extension in {input_dir}: {file_extensions}.\n"
+            "Must be one of {SUPPORTED_IMAGE_EXTENSION}."
+        )
```

Comment:
This logic should be modulized, I think.
How about this?
```
# get all images under input_dir
def get_image_files(input_dir):

# this may relieve the responsibility about checking the supported types.
def get_files(input_dir, extension):

# use both above functions
```suggestion
    # Check if all files have the same extension
    files = list(input_dir.glob("*"))
    file_extensions = set()

    for file in files:
        file_extensions.add(file.suffix)

    if len(file_extensions) != 1:
        raise ValueError(
            f"The files in {input_dir} do not have a consistent extension."
        )

    # Check if the extension of the files is supported
    unique_extension = file_extensions.pop()[1:]  # Get the unique extension existing.
    if unique_extension not in SUPPORTED_IMAGE_EXTENSION:
        raise ValueError(
            f"File extension in {input_dir}: {file_extensions}.\n"
            "Must be one of {SUPPORTED_IMAGE_EXTENSION}."
        )
```
def get_images(input_dir):
    return sum([
        get_files(input_dir, ext) for ext in SUPPORTED_IMAGE_EXTENSION
    ])
```

Reply:
@ZeroAct  I refactored the checkers. How does it look?
```python
    # Get the file extension of the input frame images
    extension = get_file_extensions(input_dir, single=True)

    # Check if the file extension is supported for image files
    if extension not in SUPPORTED_IMAGE_EXTENSION:
        raise ValueError(
            f"File extension in {input_dir}: {extension}.\n"
            f"Must be one of {SUPPORTED_IMAGE_EXTENSION}."
        )

    # Get the file extension of the output video file
    output_extension = output_path.suffix[1:]

    # Check if the file extension is supported for image files
    if output_extension not in SUPPORTED_VIDEO_EXTENSION:
        raise ValueError(
            f"Output video extension: {output_extension}.\n"
            f"Must be one of {SUPPORTED_VIDEO_EXTENSION}."
        )
```